apps/studio/src/contexts/EditorDrawerContext.tsx

```diff
@@ -1,14 +1,17 @@
+import type { IsomerComponent } from "@opengovsg/isomer-components"
 import type { Dispatch, PropsWithChildren, SetStateAction } from "react"
 import { createContext, useContext, useState } from "react"
-import { type IsomerComponent } from "@opengovsg/isomer-components"
 
+import type { SectionType } from "~/components/PageEditor/types"
 import { type DrawerState } from "~/types/editorDrawer"
 
 export interface DrawerContextType {
   currActiveIdx: number
   setCurrActiveIdx: (currActiveIdx: number) => void
   drawerState: DrawerState
   setDrawerState: (state: DrawerState) => void
+  addedBlock: Exclude<SectionType, "prose">
```

Comment:
comment: i feel like we need to rethink having so many derived states using `useState`, but out of scope of this PR.
the context should only hold the editing state, and the SSOT of the current blocks from the server.
editing state like:
- is editing (+ what ID is currently being edited, which can be then retrieved from the blocks)
- is creating

feel like these two states enough already. all the intermediate stuff like drawerState, addedBlock, pageState can theoretically be derived from the server. but nvm ba still ok for now. just very scared of desync from server state

Reply:
this is actually one of my pain points too - i think we should keep a lean context and try to either derive or use backend state. i'm not quite sure how to do it properly here, but i think this bloat is still ok for now - biggest offenders for me are `preview/savedState`. the UI only states are ok to keep